react/src/pages/analysis/Analysis.tsx

```diff
@@ -446,7 +446,6 @@ export default function Analysis() {
                         onRowUpdate: (newData, oldData) =>
                             new Promise((resolve, reject) => {
                                 // TODO: Send PATCH here for editing notes or path
-
                                 const dataUpdate = [...rows];
                                 // find the row; assume analysis_id is unique
                                 const index = dataUpdate.findIndex((row, index, obj) => {
```

Comment:
I didn't think of this before with duplicating the state block but this part should be much more clear if we go all in on the functional style like here https://github.com/ccmbioinfo/ST2020/blob/master/react/src/pages/datasets/DatasetTable.tsx#L124. Use `Array.map` to replace just the row that changed but return a different array.